### ISSUE TITLE
Is the leading boolean in `[valid? errors]` redundant?

### ISSUE THREAD

**Issue by DomKM** [michaelklishin/validateur#45]
I'm looking for a validation library and came across Validateur. It looks very nice but I don't understand the purpose of validators returning tuple of `[valid? errors]`. If a map is valid then the validator will always return `[true {}]`, right? In which case why not have validators return maps and check for success with `empty?`? It also appears that [`validation-set` creates a function that ignores leading booleans in validator tuples](https://github.com/michaelklishin/validateur/blob/0b687ddec584c163de081fc186f3b8c24c95bb87/src/cljx/validateur/validation.cljx#L539-L543).

In other words, why do validator functions and validation set functions have different contracts?

**Comment by michaelklishin** [comment]
1. Because checking for map emptiness is ugly (in my opinion).
2. Because it's too late to change such things, Validateur is about 4 years old.
